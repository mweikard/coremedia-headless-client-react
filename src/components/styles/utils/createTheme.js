// @flow
import createBreakpoints from './createBreakpoints';
import createPalette from './createPalette';
import createTypography from './createTypography';

import type { ThemeInput, ThemeOutput } from '../../../types';

const createTheme = (options: ThemeInput = {}): ThemeOutput => {
  const {
    breakpoints: breakpointsInput = {},
    palette: paletteInput = {},
    typography: typographyInput = {},
    ...rest
  } = options;

  const breakpoints = createBreakpoints(breakpointsInput);
  const palette = createPalette(paletteInput);
  const typography = createTypography(palette, breakpoints, typographyInput);

  return {
    breakpoints,
    palette,
    typography,
    overrides: {},
    ...rest,
  };
};

export default createTheme;
