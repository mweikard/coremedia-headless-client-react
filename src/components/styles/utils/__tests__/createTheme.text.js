// @flow
import createTheme from '../createTheme';

describe('createTheme()', () => {
  it(`should return matching value`, () => {
    const result = createTheme();
    expect(result).toMatchSnapshot();
  });
  it(`should return matching value using custom theme`, () => {
    const breakpoints = {
      values: {
        xs: 0,
        sm: 100,
        md: 200,
        lg: 300,
        xl: 400,
      },
      step: 10,
    };
    const palette = {
      primary: {
        color: 'rgb(20, 160, 20)',
        text: 'rgba(255, 255, 255, 1)',
      },
      secondary: {
        color: 'rgb(255, 100, 10)',
        text: 'rgba(255, 255, 255, 1)',
      },
      background: {
        color: 'rgb(250, 250, 250)',
        textPrimary: 'rgba(0, 0, 0, 1)',
        textSecondary: 'rgba(0, 0, 0, 0.8)',
        textDisabled: 'rgba(0, 0, 0, 0.5)',
      },
      surface: {
        color: 'rgb(250, 250, 250)',
        textPrimary: 'rgba(0, 0, 0, 1)',
        textSecondary: 'rgba(0, 0, 0, 0.8)',
        textDisabled: 'rgba(0, 0, 0, 0.5)',
      },
      overlay: {
        color: 'rgba(150, 150, 150, 0.9)',
      },
      additional: {
        blue: {
          color: 'rgb(10, 153, 255)',
          text: 'rgba(255, 255, 255, 1)',
        },
      },
    };
    const typography = {
      fontFamily: 'Verdana',
      fontWeightLight: 200,
      fontWeightRegular: 300,
      fontWeightMedium: 700,
      htmlFontSize: 14,
    };
    const result = createTheme({
      breakpoints,
      palette,
      typography,
      other: {
        value: 'text',
      },
    });
    expect(result).toMatchSnapshot();
  });
});
