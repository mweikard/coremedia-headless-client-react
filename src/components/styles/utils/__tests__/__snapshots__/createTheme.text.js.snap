// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTheme() should return matching value 1`] = `
Object {
  "breakpoints": Object {
    "between": [Function],
    "exact": [Function],
    "keys": Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ],
    "max": [Function],
    "min": [Function],
    "values": Object {
      "lg": 1280,
      "md": 992,
      "sm": 768,
      "xl": 1920,
      "xs": 0,
    },
  },
  "palette": Object {
    "additional": Object {},
    "background": Object {
      "color": "#fff",
      "textDisabled": "rgba(0,0,0,0.38)",
      "textPrimary": "rgba(0,0,0,0.87)",
      "textSecondary": "rgba(0,0,0,0.6)",
    },
    "common": Object {
      "black": "#000",
      "white": "#fff",
    },
    "overlay": Object {
      "color": "rgba(173,173,173,0.9)",
    },
    "primary": Object {
      "color": "#0a84ff",
      "text": "rgba(255,255,255,0.87)",
    },
    "secondary": Object {
      "color": "#ed00b5",
      "text": "rgba(255,255,255,0.87)",
    },
    "surface": Object {
      "color": "#fff",
      "textDisabled": "rgba(0,0,0,0.38)",
      "textPrimary": "rgba(0,0,0,0.87)",
      "textSecondary": "rgba(0,0,0,0.6)",
    },
  },
  "typography": Object {
    "body1": Object {
      "color": "rgba(0,0,0,0.87)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1rem",
      "fontWeight": 400,
      "lineHeight": "1.3125em",
    },
    "body2": Object {
      "color": "rgba(0,0,0,0.87)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "0.875rem",
      "fontWeight": 400,
      "lineHeight": "1.2857142857142858em",
    },
    "button": Object {
      "color": "rgba(0,0,0,0.87)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1rem",
      "fontWeight": 500,
      "textTransform": "none",
    },
    "caption": Object {
      "color": "rgba(0,0,0,0.6)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "lineHeight": "1.3333333333333333em",
      "textTransform": "none",
    },
    "fontFamily": "Helvetica, Arial, sans-serif",
    "fontWeightLight": 300,
    "fontWeightMedium": 500,
    "fontWeightRegular": 400,
    "htmlFontSize": 16,
    "title1": Object {
      "color": "rgba(0,0,0,0.6)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "2.125rem",
      "fontWeight": 300,
      "lineHeight": "1.2647058823529411em",
      "textTransform": "uppercase",
    },
    "title2": Object {
      "color": "rgba(0,0,0,0.6)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1.875rem",
      "fontWeight": 300,
      "lineHeight": "1.1333333333333333em",
      "textTransform": "uppercase",
    },
    "title3": Object {
      "color": "rgba(0,0,0,0.6)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1.625rem",
      "fontWeight": 300,
      "lineHeight": "1.1538461538461537em",
      "textTransform": "uppercase",
    },
    "title4": Object {
      "color": "rgba(0,0,0,0.6)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1.4375rem",
      "fontWeight": 400,
      "lineHeight": "1.2608695652173914em",
      "textTransform": "uppercase",
    },
    "title5": Object {
      "color": "rgba(0,0,0,0.87)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1.25rem",
      "fontWeight": 400,
      "lineHeight": "1.3em",
    },
    "title6": Object {
      "color": "rgba(0,0,0,0.87)",
      "fontFamily": "Helvetica, Arial, sans-serif",
      "fontSize": "1.125rem",
      "fontWeight": 400,
      "lineHeight": "1.2777777777777777em",
    },
  },
}
`;

exports[`createTheme() should return matching value using custom theme 1`] = `
Object {
  "breakpoints": Object {
    "between": [Function],
    "exact": [Function],
    "keys": Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ],
    "max": [Function],
    "min": [Function],
    "values": Object {
      "lg": 300,
      "md": 200,
      "sm": 100,
      "xl": 400,
      "xs": 0,
    },
  },
  "other": Object {
    "value": "text",
  },
  "palette": Object {
    "additional": Object {
      "blue": Object {
        "color": "rgb(10, 153, 255)",
        "text": "rgba(255, 255, 255, 1)",
      },
    },
    "background": Object {
      "color": "rgb(250, 250, 250)",
      "textDisabled": "rgba(0, 0, 0, 0.5)",
      "textPrimary": "rgba(0, 0, 0, 1)",
      "textSecondary": "rgba(0, 0, 0, 0.8)",
    },
    "common": Object {
      "black": "#000",
      "white": "#fff",
    },
    "overlay": Object {
      "color": "rgba(150, 150, 150, 0.9)",
    },
    "primary": Object {
      "color": "rgb(20, 160, 20)",
      "text": "rgba(255, 255, 255, 1)",
    },
    "secondary": Object {
      "color": "rgb(255, 100, 10)",
      "text": "rgba(255, 255, 255, 1)",
    },
    "surface": Object {
      "color": "rgb(250, 250, 250)",
      "textDisabled": "rgba(0, 0, 0, 0.5)",
      "textPrimary": "rgba(0, 0, 0, 1)",
      "textSecondary": "rgba(0, 0, 0, 0.8)",
    },
  },
  "typography": Object {
    "body1": Object {
      "color": "rgba(0, 0, 0, 1)",
      "fontFamily": "Verdana",
      "fontSize": "1.1428571428571428rem",
      "fontWeight": 300,
      "lineHeight": "1.3125em",
    },
    "body2": Object {
      "color": "rgba(0, 0, 0, 1)",
      "fontFamily": "Verdana",
      "fontSize": "1rem",
      "fontWeight": 300,
      "lineHeight": "1.2857142857142858em",
    },
    "button": Object {
      "color": "rgba(0, 0, 0, 1)",
      "fontFamily": "Verdana",
      "fontSize": "1.1428571428571428rem",
      "fontWeight": 700,
      "textTransform": "none",
    },
    "caption": Object {
      "color": "rgba(0, 0, 0, 0.8)",
      "fontFamily": "Verdana",
      "fontSize": "0.8571428571428571rem",
      "fontWeight": 300,
      "lineHeight": "1.3333333333333333em",
      "textTransform": "none",
    },
    "fontFamily": "Verdana",
    "fontWeightLight": 200,
    "fontWeightMedium": 700,
    "fontWeightRegular": 300,
    "htmlFontSize": 14,
    "title1": Object {
      "color": "rgba(0, 0, 0, 0.8)",
      "fontFamily": "Verdana",
      "fontSize": "2.4285714285714284rem",
      "fontWeight": 200,
      "lineHeight": "1.2647058823529411em",
      "textTransform": "uppercase",
    },
    "title2": Object {
      "color": "rgba(0, 0, 0, 0.8)",
      "fontFamily": "Verdana",
      "fontSize": "2.142857142857143rem",
      "fontWeight": 200,
      "lineHeight": "1.1333333333333333em",
      "textTransform": "uppercase",
    },
    "title3": Object {
      "color": "rgba(0, 0, 0, 0.8)",
      "fontFamily": "Verdana",
      "fontSize": "1.8571428571428572rem",
      "fontWeight": 200,
      "lineHeight": "1.1538461538461537em",
      "textTransform": "uppercase",
    },
    "title4": Object {
      "color": "rgba(0, 0, 0, 0.8)",
      "fontFamily": "Verdana",
      "fontSize": "1.6428571428571428rem",
      "fontWeight": 300,
      "lineHeight": "1.2608695652173914em",
      "textTransform": "uppercase",
    },
    "title5": Object {
      "color": "rgba(0, 0, 0, 1)",
      "fontFamily": "Verdana",
      "fontSize": "1.4285714285714286rem",
      "fontWeight": 300,
      "lineHeight": "1.3em",
    },
    "title6": Object {
      "color": "rgba(0, 0, 0, 1)",
      "fontFamily": "Verdana",
      "fontSize": "1.2857142857142858rem",
      "fontWeight": 300,
      "lineHeight": "1.2777777777777777em",
    },
  },
}
`;
